tests/test_campaign.cairo

```diff
@@ -201,6 +201,62 @@ fn test_withdraw() {
     assert(token_giver.get_available_withdrawal(campaign_address) == 0, 'withdrawal failed');
 }
 
+
+#[test]
+#[fork("Mainnet")]
+fn test_create_campaign() {
+    let (token_giver_address, _) = __setup__();
+    let token_giver = ICampaignDispatcher { contract_address: token_giver_address };
+
+    // create campaign
+    start_cheat_caller_address(token_giver_address, RECIPIENT());
+    let created_campaign_address = token_giver
+        .create_campaign(REGISTRY_HASH(), IMPLEMENTATION_HASH(), SALT());
+    stop_cheat_caller_address(token_giver_address);
+
+    // get campagin
+    start_cheat_caller_address(token_giver_address, RECIPIENT());
+    let campaign = token_giver.get_campaign(created_campaign_address);
+
+    let campaign_address = campaign.campaign_address;
+
+    stop_cheat_caller_address(token_giver_address);
+
+    assert(created_campaign_address == campaign_address, 'create campaign failed');
+}
+
+
+#[test]
+#[fork("Mainnet")]
+fn test_create_campaign_event_emission() {
+    let (token_giver_address, _) = __setup__();
+    let token_giver = ICampaignDispatcher { contract_address: token_giver_address };
+
+    // create campaign
+    start_cheat_caller_address(token_giver_address, RECIPIENT());
+    let created_campaign_address = token_giver
+        .create_campaign(REGISTRY_HASH(), IMPLEMENTATION_HASH(), SALT());
+    stop_cheat_caller_address(token_giver_address);
+
+    // get campagin
+    start_cheat_caller_address(token_giver_address, RECIPIENT());
+    let campaign = token_giver.get_campaign(created_campaign_address);
+
+    let campaign_address = campaign.campaign_address;
```

Comment:
Don't define them like this  (delete them)
```

  let campaign_address = campaign.campaign_address;
    let campaign_owner = campaign.campaign_owner;
    let token_id = campaign.token_id;
```

use them directly i.e campaign.token_id, campaign.campaign_owner and campaign.campaign_address